modules/blockbuilder/tenant_store.go

```diff
@@ -230,6 +232,27 @@ func (s *tenantStore) CutIdle(since time.Time, immediate bool) error {
 	return s.cutHeadBlock(false)
 }
 
+func (s *tenantStore) adjustTimeRangeForSlack(startSectionTime time.Time, start, end uint32) (uint32, uint32) {
+	startOfRange := uint32(startSectionTime.Add(-s.headBlock.IngestionSlack()).Unix())
+	endOfRange := uint32(startSectionTime.Add(s.headBlock.IngestionSlack()).Unix())
+
+	warn := false
+	if start < startOfRange {
+		warn = true
+		start = uint32(startSectionTime.Unix())
+	}
+	if end > endOfRange || end < start {
+		warn = true
+		end = uint32(startSectionTime.Unix())
+	}
+
+	if warn {
+		dataquality.WarnOutsideIngestionSlack(s.headBlock.BlockMeta().TenantID)
+	}
+
+	return start, end
+}
+
```

Comment:
We have now 4 implementations of the same function. IMO this should be moved to a different pkg and reused.

Reply:
I've moved it to a common place